Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -161,17 +166,22 @@ - (void)characterTyped:(unichar)c asInsertedCharacter:(BOOL)inserted previousCha
                 }
             }
             else if ((currentCharacterType == CharacterTypeControlCharacter)
-                     && (previousCharacterType == CharacterTypeWhitespace || previousCharacter == 0)) {
-                if (self.charactersSinceLastWhitespace == 0) {
+                     && (previousCharacterType == CharacterTypeSeparator || previousCharacter == 0)) {
+                if (self.charactersSinceLastWhitespace == 0 || previousCharacterType == CharacterTypeSeparator) {
```

Comment:
why do we need to check characters since if we're already confirming that the previous characterType is a separator

Reply:
Changed the `charactersSinceLastWhitespace` check to "`previousCharacter` == 0" check. As we are setting previous character type to normal by default (as it would be true for most of the cases), previous character type would be set to normal for the beginning of a line and previous character type check would fail in that case which is why we need the previous character check.